File: Exercises/Codewar/Haskell/population.hs
module Codewars.G964.Arge where

nbYear :: Int -> Double -> Int -> Int -> Int
nbYear p0 percent aug p
 | pop >= p    = 1
 | otherwise   = 1 + nbYear pop percent aug p
 where pop = p0 + floor ((fromIntegral p0) * percent / 100.0) + aug

File: Exercises/Codewar/C++/population.cxx
struct Arge {
	static int nbYear(int p0, double percent, int aug, int p)
	{
		int years = 0;
		auto next_pop = [](int p0, double perc, int aug) {
			return int(p0 * (1+perc/100.0)) + aug;
		};
		for (; p0 < p; ++years)
			p0 = next_pop(p0, percent, aug);
		return years;
	}
};
